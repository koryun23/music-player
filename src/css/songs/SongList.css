.song-list {
    width: 100%;
    margin-top: 10px;
}

.song-row {
    display: grid;
    grid-template-columns: 5% 30% 30% 20% 15%;
    border-bottom: 1px solid rgb(229, 229, 229);
}

.top-row {
    display: grid;
    grid-template-columns: 5% 30% 30% 20% 15%;
    gap: 1px;
}

.top-col {
    background-color:rgb(229, 229, 229);
    padding: 15px;
}
.col {
    padding: 10px;
}

.song-play-and-move {
    display: flex;
}

.dots {
    float: left;
    background-color:white;
    border: 0px;
}

.play {
    float: left;
    background-color:white;
    border: 0px;
}

.additional-options {
    display: grid;
    grid-template-columns: auto auto auto auto;
}

.option {
    background-color: white;
    border: 0px;
}