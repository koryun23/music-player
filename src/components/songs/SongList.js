import "../../css/songs/SongList.css";
import { React, useState } from "react";
import SongRow from "./SongRow";
export default function SongList(props) {
    
    return (
        <table className="table">
            <thead className="thead-light">
                <tr>
                    <th scope="col"></th>
                    <th scope="col">Song Name</th>
                    <th scope="col">Artist Name</th>
                    <th scope="col">Track</th>
                    <th scope="col"></th>

                    <th scope="col"></th>
                    <th scope="col"></th>
                    <th scope="col"></th>
                    <th scope="col"></th>
                </tr>
            </thead>
            <tbody>
                {props.songList.map(song => (
                    <SongRow songName={song.songName}
                            artistName={song.artistName} 
                            trackNumber={song.trackNumber} 
                            fileName={song.fileName}
                            onPlaySingle={(fileName) => props.onPlaySingle(fileName)}
                            isPlaying={props.isPlaying && props.fileName == song.fileName} 
                            isFavoriteSong={props.favoriteSongList.includes(song.songName)} 
                            onAddToFavorites={() => props.onAddToFavorites(song.songName)}
                            onRemoveFromFavorites={() => props.onRemoveFromFavorites(song.songName)} 
                            onStopPlaying={() => props.onStopPlaying(song.fileName)}
                            onPlayNextSong={() => props.onPlayNextSong(song.fileName)}
                            onRemoveFromPlaylist={() => props.onRemoveFromPlaylist(song.fileName)}
                            onRemoveFromQueue={() => {props.onRemoveFromQueue(song.fileName)}}/>
                ))}
            </tbody>
        </table>
    );
}