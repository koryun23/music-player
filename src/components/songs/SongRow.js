import { React, useState} from "react";
import "../../css/songs/SongList.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCaretDown, faCheck, faForward, faGrip, faHeart, faPause, faPlay, faReply, faShare } from "@fortawesome/free-solid-svg-icons";
import { Button, ButtonGroup } from "react-bootstrap";

export default function SongRow(props) {

    const [songName, setSongName] = useState(props.songName);
    const [artistName, setArtistName] = useState(props.artistName);
    const [trackNumber, setTrackNumber] = useState(props.trackNumber);
    const [fileName, setFileName] = useState(props.fileName);

    const row = (
        <tr>
            <th scope="row">
                <button className="dots">
                    <FontAwesomeIcon icon={faGrip}/>
                </button>
                <button className="play" onClick={() => props.onPlaySingle(fileName)}>
                    {
                        (props.isPlaying && <FontAwesomeIcon icon={faPause}/>) ||
                        (<FontAwesomeIcon icon={faPlay}/>)
                    }
                </button>
            </th>
            <td>
                {props.songName}
            </td>
            <td>
                {props.artistName}
            </td>
            <td>
                {props.trackNumber}
            </td>
            <td>
                    <button className="option">
                        <FontAwesomeIcon icon={faHeart}/>
                    </button>
                    <button className="option">
                        <FontAwesomeIcon icon={faCheck}/>
                    </button>
                    <button className="option">
                        <FontAwesomeIcon icon={faShare}/>
                    </button>
                    <button className="option">
                        <FontAwesomeIcon icon={faCaretDown} />
                    </button>
            </td>
        </tr>
    );
    return row;
} 