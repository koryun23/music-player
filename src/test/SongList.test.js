import renderer from "react-test-renderer";
import SongRow from "../components/songs/SongList";
import SongList from "../components/songs/SongList";

it('song list renders correctly', () => {
    const songList = [
        {songName: "aaa", artistName: "bbb", fileName: "ccc", trackNumber: 1},
        {songName: "aaaa", artistName: "bbbb", fileName: "cccc", trackNumber: 2},
        {songName: "aa", artistName: "bb", fileName: "cc", trackNumber: 3},
        {songName: "a", artistName: "b", fileName: "c", trackNumber: 4},

    ]
    const tree = renderer.create(<SongList songList={songList}
                                           favoriteSongList={songList} 
                                          />).toJSON();
    expect(tree).toMatchSnapshot();
})