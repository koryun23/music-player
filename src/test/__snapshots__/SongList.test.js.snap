// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`song list renders correctly 1`] = `
<table
  className="table table-hover"
>
  <thead
    className="thead-light"
  >
    <tr>
      <th
        scope="col"
      />
      <th
        scope="col"
      >
        Song Name
      </th>
      <th
        scope="col"
      >
        Artist Name
      </th>
      <th
        scope="col"
      >
        Track
      </th>
      <th
        scope="col"
      />
      <th
        scope="col"
      />
      <th
        scope="col"
      />
    </tr>
  </thead>
  <tbody>
    <tr>
      <th
        scope="row"
      >
        <button
          className="play"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-play "
            data-icon="play"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </th>
      <td>
        aaa
      </td>
      <td>
        bbb
      </td>
      <td>
        1
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-heart "
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-stop "
            data-icon="stop"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.7 28.7 64 64 64H320c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-forward "
            data-icon="forward"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M52.5 440.6c-9.5 7.9-22.8 9.7-34.1 4.4S0 428.4 0 416V96C0 83.6 7.2 72.3 18.4 67s24.5-3.6 34.1 4.4L224 214.3V256v41.7L52.5 440.6zM256 352V256 128 96c0-12.4 7.2-23.7 18.4-29s24.5-3.6 34.1 4.4l192 160c7.3 6.1 11.5 15.1 11.5 24.6s-4.2 18.5-11.5 24.6l-192 160c-9.5 7.9-22.8 9.7-34.1 4.4s-18.4-16.6-18.4-29V352z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        <button
          className="play"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-play "
            data-icon="play"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </th>
      <td>
        aaaa
      </td>
      <td>
        bbbb
      </td>
      <td>
        2
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-heart "
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-stop "
            data-icon="stop"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.7 28.7 64 64 64H320c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-forward "
            data-icon="forward"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M52.5 440.6c-9.5 7.9-22.8 9.7-34.1 4.4S0 428.4 0 416V96C0 83.6 7.2 72.3 18.4 67s24.5-3.6 34.1 4.4L224 214.3V256v41.7L52.5 440.6zM256 352V256 128 96c0-12.4 7.2-23.7 18.4-29s24.5-3.6 34.1 4.4l192 160c7.3 6.1 11.5 15.1 11.5 24.6s-4.2 18.5-11.5 24.6l-192 160c-9.5 7.9-22.8 9.7-34.1 4.4s-18.4-16.6-18.4-29V352z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        <button
          className="play"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-play "
            data-icon="play"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </th>
      <td>
        aa
      </td>
      <td>
        bb
      </td>
      <td>
        3
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-heart "
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-stop "
            data-icon="stop"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.7 28.7 64 64 64H320c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-forward "
            data-icon="forward"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M52.5 440.6c-9.5 7.9-22.8 9.7-34.1 4.4S0 428.4 0 416V96C0 83.6 7.2 72.3 18.4 67s24.5-3.6 34.1 4.4L224 214.3V256v41.7L52.5 440.6zM256 352V256 128 96c0-12.4 7.2-23.7 18.4-29s24.5-3.6 34.1 4.4l192 160c7.3 6.1 11.5 15.1 11.5 24.6s-4.2 18.5-11.5 24.6l-192 160c-9.5 7.9-22.8 9.7-34.1 4.4s-18.4-16.6-18.4-29V352z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        <button
          className="play"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-play "
            data-icon="play"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </th>
      <td>
        a
      </td>
      <td>
        b
      </td>
      <td>
        4
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-heart "
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-stop "
            data-icon="stop"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.7 28.7 64 64 64H320c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-forward "
            data-icon="forward"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M52.5 440.6c-9.5 7.9-22.8 9.7-34.1 4.4S0 428.4 0 416V96C0 83.6 7.2 72.3 18.4 67s24.5-3.6 34.1 4.4L224 214.3V256v41.7L52.5 440.6zM256 352V256 128 96c0-12.4 7.2-23.7 18.4-29s24.5-3.6 34.1 4.4l192 160c7.3 6.1 11.5 15.1 11.5 24.6s-4.2 18.5-11.5 24.6l-192 160c-9.5 7.9-22.8 9.7-34.1 4.4s-18.4-16.6-18.4-29V352z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`song row renders correctly 1`] = `
<table
  className="table table-hover"
>
  <thead
    className="thead-light"
  >
    <tr>
      <th
        scope="col"
      />
      <th
        scope="col"
      >
        Song Name
      </th>
      <th
        scope="col"
      >
        Artist Name
      </th>
      <th
        scope="col"
      >
        Track
      </th>
      <th
        scope="col"
      />
      <th
        scope="col"
      />
      <th
        scope="col"
      />
    </tr>
  </thead>
  <tbody>
    <tr>
      <th
        scope="row"
      >
        <button
          className="play"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-play "
            data-icon="play"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </th>
      <td>
        aaa
      </td>
      <td>
        bbb
      </td>
      <td>
        1
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-heart "
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-stop "
            data-icon="stop"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.7 28.7 64 64 64H320c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-forward "
            data-icon="forward"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M52.5 440.6c-9.5 7.9-22.8 9.7-34.1 4.4S0 428.4 0 416V96C0 83.6 7.2 72.3 18.4 67s24.5-3.6 34.1 4.4L224 214.3V256v41.7L52.5 440.6zM256 352V256 128 96c0-12.4 7.2-23.7 18.4-29s24.5-3.6 34.1 4.4l192 160c7.3 6.1 11.5 15.1 11.5 24.6s-4.2 18.5-11.5 24.6l-192 160c-9.5 7.9-22.8 9.7-34.1 4.4s-18.4-16.6-18.4-29V352z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        <button
          className="play"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-play "
            data-icon="play"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </th>
      <td>
        aaaa
      </td>
      <td>
        bbbb
      </td>
      <td>
        2
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-heart "
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-stop "
            data-icon="stop"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.7 28.7 64 64 64H320c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-forward "
            data-icon="forward"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M52.5 440.6c-9.5 7.9-22.8 9.7-34.1 4.4S0 428.4 0 416V96C0 83.6 7.2 72.3 18.4 67s24.5-3.6 34.1 4.4L224 214.3V256v41.7L52.5 440.6zM256 352V256 128 96c0-12.4 7.2-23.7 18.4-29s24.5-3.6 34.1 4.4l192 160c7.3 6.1 11.5 15.1 11.5 24.6s-4.2 18.5-11.5 24.6l-192 160c-9.5 7.9-22.8 9.7-34.1 4.4s-18.4-16.6-18.4-29V352z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        <button
          className="play"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-play "
            data-icon="play"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </th>
      <td>
        aa
      </td>
      <td>
        bb
      </td>
      <td>
        3
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-heart "
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-stop "
            data-icon="stop"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.7 28.7 64 64 64H320c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-forward "
            data-icon="forward"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M52.5 440.6c-9.5 7.9-22.8 9.7-34.1 4.4S0 428.4 0 416V96C0 83.6 7.2 72.3 18.4 67s24.5-3.6 34.1 4.4L224 214.3V256v41.7L52.5 440.6zM256 352V256 128 96c0-12.4 7.2-23.7 18.4-29s24.5-3.6 34.1 4.4l192 160c7.3 6.1 11.5 15.1 11.5 24.6s-4.2 18.5-11.5 24.6l-192 160c-9.5 7.9-22.8 9.7-34.1 4.4s-18.4-16.6-18.4-29V352z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        <button
          className="play"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-play "
            data-icon="play"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </th>
      <td>
        a
      </td>
      <td>
        b
      </td>
      <td>
        4
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-heart "
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-stop "
            data-icon="stop"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.7 28.7 64 64 64H320c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
      <td>
        <button
          className="option"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-forward "
            data-icon="forward"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M52.5 440.6c-9.5 7.9-22.8 9.7-34.1 4.4S0 428.4 0 416V96C0 83.6 7.2 72.3 18.4 67s24.5-3.6 34.1 4.4L224 214.3V256v41.7L52.5 440.6zM256 352V256 128 96c0-12.4 7.2-23.7 18.4-29s24.5-3.6 34.1 4.4l192 160c7.3 6.1 11.5 15.1 11.5 24.6s-4.2 18.5-11.5 24.6l-192 160c-9.5 7.9-22.8 9.7-34.1 4.4s-18.4-16.6-18.4-29V352z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
